### PR TITLE
Cherry pick MM-53842, MM-54275 to v7.8

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54275
https://mattermost.atlassian.net/browse/MM-53842

#### Release Note

```release-note
Replaced Gfycat with Giphy in gif picker.
```

### CODE REVIEW

**Review by marianunez — APPROVED** [review]

**Comment by amyblais** [comment]
@M-ZubairAhmed There are failing tests here.

File: config/client.go [hunk 1]
```diff
@@ -80,6 +80,7 @@ func GenerateClientConfig(c *model.Config, telemetryID string, license *model.Li
 	props["EnableGifPicker"] = strconv.FormatBool(*c.ServiceSettings.EnableGifPicker)
 	props["GfycatApiKey"] = *c.ServiceSettings.GfycatAPIKey
 	props["GfycatApiSecret"] = *c.ServiceSettings.GfycatAPISecret
+	props["GiphySdkKey"] = *c.ServiceSettings.GiphySdkKey
```
Comment (agnivade) [inline config/client.go:83]: If this is only for 7.8, do we even want a config setting? i.e. do we see a case where customers might want to use their own key?

**Comment by agnivade** [comment]
We want to port over the change we are doing for the master branch right?

**Comment by M-ZubairAhmed** [comment]
> We want to port over the change we are doing for the master branch right?

Yes they are now ported over to this PR

File: services/telemetry/telemetry.go [hunk 2]
```diff
@@ -389,6 +389,7 @@ func (ts *TelemetryService) trackConfig() {
 		"enable_gif_picker":                                       *cfg.ServiceSettings.EnableGifPicker,
 		"gfycat_api_key":                                          isDefault(*cfg.ServiceSettings.GfycatAPIKey, model.ServiceSettingsDefaultGfycatAPIKey),
 		"gfycat_api_secret":                                       isDefault(*cfg.ServiceSettings.GfycatAPISecret, model.ServiceSettingsDefaultGfycatAPISecret),
+		"giphy_sdk_key":                                           *cfg.ServiceSettings.GiphySdkKey,
```
Comment (marianunez) [inline services/telemetry/telemetry.go:392]: Did we do this one in the other branches?

**Review by marianunez — APPROVED** [review]

File: services/telemetry/telemetry.go (hunk 2 above)
Comment (M-ZubairAhmed) [reply services/telemetry/telemetry.go:392]: I added this here because CI complained, in others they didn't so.

Comment (marianunez) [reply services/telemetry/telemetry.go:392]: Oh maybe this was added with the original implementation in the monorepo PR

Comment (agnivade) [reply services/telemetry/telemetry.go:392]: You can add  `// telemetry: none` as a comment beside the `GiphySdkKey` declaration. I don't think we want to upload the key for telemetry.

**Review by agnivade — APPROVED** [review]

**Pull request merged**